pkg/ingress/model_build_load_balancer.go

```diff
@@ -255,7 +255,18 @@ func (t *defaultModelBuildTask) buildLoadBalancerSecurityGroups(ctx context.Cont
 		if err != nil {
 			return nil, err
 		}
-		return []core.StringToken{sg.GroupID()}, nil
+		sgs := []core.StringToken{sg.GroupID()}
+
```

Comment:
1. i'm wondering whether we should make IngressSuffixManageSecurityGroupRules generic for auto-generated SG case as well. e.g. If it's explicitly specified as false, then we still generate the frontend SG but don't generate backendSG and don't manage the worker node SG rules. 
2. it would be good to make the logic of handling backendSG for both `auto-generate frontend SG` and `manually specified frontend SG` in a common generic place here.
3. it would be good to handle the knob of whether enable backendSG feature here instead of inside backendSGProvider. i.e. we store the backendSG into the modelBuildTask, for 'auto-generated frontend SG` case, it will be either the managed SG or the backendSG depends on whether backendSG feature is enabled or not. Instead of handling it in the TGB model build task. (this is more flexible by leaking less implementation detail to TGB model build tasks)